src/directives/internal/class.js

```diff
@@ -69,7 +85,15 @@ function stringToObject (value) {
 }
 
 function contains (value, key) {
-  return isArray(value)
-    ? value.indexOf(key) > -1
-    : hasOwn(value, key)
+  if (isArray(value)) {
+    for (var i = 0; i < value.length; i++) {
+      if (isPlainObject(value[i])) {
+        return hasOwn(value[i], key)
+      } else {
+        return value[i] === key
```

Comment:
same here: should be a conditional return true.

We can also cache `value[i]` here.